Sources/Authentication/Header/BasicHeaderValueAuthorization.swift

```diff
@@ -0,0 +1,11 @@
+
+/// A basic authorization type that can be
+/// used with any HTTP header authorization
+/// strategy.
+/// See `BearerAuthorization` for a common
+/// alternative.
+public struct BasicHeaderValueAuthorization: HeaderValueAuthorization {
```

Comment:
Do we need this as a separate type from `BearerAuthorization`?

Reply:
Originally I created `BasicHeaderValueAuthorization` simply because it was semantically confusing to me that an authentication strategy unrelated to Bearer Authorization would depend on a authorization token/value named after that strategy. I also considered just using `String` (or creating a `typealias` of `String) because `BasicHeaderValueAuthorization` does not need to actually do anything.

However, then `BearerAuthorization` became the basis for differentiating `BearerAuthenticatable` from the more general-purpose `HeaderValueAuthenticatable`. That locked in `HeaderValueAuthenticatable` as its own type. I don't think that is necessary, so I am happy to revisit this approach to differentiating the two `Authenticatable` protocols. I admit I've grown fond of the explicit distinction between the two concrete `HeaderValueAuthorization` types, though.

Do you think it is worth me revisiting the strategy I used for differentiating `BearerAuthenticatable` from `HeaderValueAuthenticatable`?

Reply:
I'm not sure I understand what you mean by "differentiating" them, but I think it makes sense for `BearerAuthenticatable` to refine `HeaderValueAuthenticatable` by adding a default impl of `authorization(from: HTTPHeaders)`. So if you only conform to `HeaderValueAuthenticatable`, you must implement how the token is fetched from headers. If you conform to `BearerAuthenticatable`, you get that method implemented by default, but you can still override it. 

Regarding the return type of `authorization(from: HTTPHeaders)`, I think that would make sense as a `String`. The idea is that you are comparing the property on the model to the property from the header, and the `authToken` property is a `String`. So having parity there would be nice I think. I might even go a step further and rename `authorization(from: HTTPHeaders)` to be `authToken(from: HTTPHeaders)` so that the relation is clear.

Reply:
The "differentiate" comment stemmed from a misunderstanding I had about Swift's protocol inheritance so never mind that; Funny, I've been deep down the "generics" rabbit hole before but I start to lose sight of the easy things we can do because I've spent so much time finding the edges of Swift's capabilities.

On a related note, great suggestions all-around. Using `String` does nothing but simplify the implementation. I guess I was hung up on `BearerAuthorization` existing even though it is also nothing but a `String`.